app/models/terminal_executor.rb

```diff
@@ -131,7 +134,19 @@ def script(commands)
   end
 
   def print_and_execute(c, resolve: true)
-    "echo » #{c.shellescape}\n#{resolve ? resolve_secrets(c) : c}"
+    print_and_execute_hidden_command(c) ||
+      print_and_execute_raw(c, resolve ? resolve_secrets(c) : c)
+  end
+
+  def print_and_execute_hidden_command(c)
+    # Replace `hidden://abcd` up until the first doublequote with HIDDEN_TXT
```

Comment:
prefer "why it does it" over "what it does"
`# hides secrets by replacing lines like export FOO="hidden://secret" into export FOO="HIDDEN"`